design/specs/listing-views.md

```diff
@@ -1,27 +1,120 @@
+- An error message is displayed indicating to the user that the loading of objects failed.
+- The error notice contains details about the error and actions, if available, that the user can perform to correct the problem.
+
+### Scenario 3: User encounters an error while opening a view in a currently open schema
+
+#### Scenario 3a: From the schema explorer sidebar
+
+- The user opens the schema explorer by navigating to it via the sidebar or by URL pointing to that object.
+- The object fails to load, and the system displays an error notice.
+
+#### Scenario 3b: From an open tab
+
+- The user navigates to a tab, which represents a previously open view.
+- The object fails to load, and the system displays an error notice.
+
+### Scenario 4: User searches a view by name
+
+#### Scenario 4a: From the schema explorer view's sidebar
+
+- The user opens the schema explorer by navigating to it via the sidebar or by URL pointing to that object.
+- The user enters a search term in the search input box located at the top of the sidebar list.
+- The list is filtered to show objects whose name matches the search term.
+- The list contains metadata associated with the objects like record and column count.
+  - If no results are found, the system displays a notice to indicate the lack of results and potential causes.
+- Users can clear the search by clearing the input box content or clicking on the `Clear Search` link displayed along with the results summary.
+
+## Views
+
+### Schema Details Page
+
+The schema details page will be shown when a schema is open and notables are selected. It will contain the schema's name, the total number of tables and views, and additional sections that we can use to present the user with relevant information or links. Examples of content that could go into this section are:
+
+- Recently opened views or tables
+- Activity summary for recent events
+- Getting started options for common actions
+
+## Components
+
+### Schema Explorer Sidebar
+
+The schema explorer sidebar contains the entire list of objects for any schema available in Mathesar. The user can open any object from the sidebar by clicking on its navigation link. The sidebar also includes an input for searching through the listed objects.
+
+### Sidebar Tabs
+
+Sidebar tabs will allow users to filter the sidebar's contents to show only a particular object type. When set to ' All ', the content will include both tables and views. Setting the tabs to `Tables` will display only table objects, while `Views` will display only view objects.
+
+## Other Considerations
+
+### Regular vs. Materialized Views
+
+Regular views need to be differentiated from materialized views as the use cases might differ. However, we don't want to create a rigid distinction between both as it could confuse users being introduced to the concept of views. For this purpose, an indicator, such as the letter 'M' can be added next to the view icon to show that it is of the type 'Materialized'.
```

Comment:
What does "can be added" mean here? Should it be added or not?

Reply:
@kgodey I did add an indicator to the design. I believe it should be added, I will update the spec to remove the ambiguity.